backend/src/match/match.service.ts

```diff
@@ -1,24 +1,30 @@
 import { Injectable } from "@nestjs/common";
+import { nanoid } from "nanoid";
+import { InjectPinoLogger, PinoLogger } from "nestjs-pino";
 
 import { RedisService } from "src/redis/redis.service";
 
-interface Match {
-  // TODO: Create a room ID with password
-  result: { userId: number; socketId: string }[];
-}
+import { Match } from "~shared/types/api/match.dto";
 
 @Injectable()
 export class MatchService {
   private static readonly NAMESPACE = "Match";
   private static readonly EXPIRATION_TIME = 30;
 
-  constructor(private redisService: RedisService) {}
+  constructor(
+    @InjectPinoLogger()
+    private readonly logger: PinoLogger,
+    private readonly redisService: RedisService,
+  ) {}
 
   async searchMatch(
```

Comment:
Let's leave a TODO comment here to prevent users from matching themselves if they have multiple tabs open.